tango/step.py

```diff
@@ -354,18 +359,31 @@ def _run_with_work_dir(self, workspace: "Workspace", **kwargs) -> T:
             self.work_dir_for_run = Path(dir_for_cleanup.name)
 
         try:
-            if self.cache_results:
-                workspace.step_starting(self)
+            kwargs = self._replace_steps_with_results(self.kwargs, workspace)
+
+            if needed_by:
+                cli_logger.info(
+                    '[blue]\N{black circle} Starting step [bold]"%s"[/] (needed by "%s")...[/]',
+                    self.name,
+                    needed_by.name,
+                )
+            else:
+                cli_logger.info(
+                    '[blue]\N{black circle} Starting step [bold]"%s"[/]...[/]',
+                    self.name,
+                )
+
+            workspace.step_starting(self)
+
             try:
                 result = self.run(**kwargs)
-                if self.cache_results:
-                    result = workspace.step_finished(self, result)
+                result = workspace.step_finished(self, result)
+                cli_logger.info(f'[green]\N{check mark} Finished step [bold]"{self.name}"[/][/]')
                 return result
             except BaseException as e:
                 # TODO (epwalsh): do we want to handle KeyboardInterrupts differently?
                 # Maybe have a `workspace.step_interrupted()` method?
```

Comment:
No. Why would we?

Reply:
To distinguish between steps that failed and steps that were interrupted. But I don't care enough to argue. 5f1ac38